core/auth/src/main/java/software/amazon/awssdk/auth/credentials/internal/SystemSettingsCredentialsProvider.java

```diff
@@ -67,8 +86,16 @@ public AwsCredentials resolveCredentials() {
                                     .build();
         }
 
-        return sessionToken == null ? AwsBasicCredentials.create(accessKey, secretKey)
-                                    : AwsSessionCredentials.create(accessKey, secretKey, sessionToken);
+        if (webIdentityTokenFile != null && roleArn != null) {
+            String webIdentityToken = WebIdentityCredentialsUtil.resolveWebIdentityToken(Paths.get(webIdentityTokenFile));
+            WebIdentityTokenCredentialsProviderFactory factory = WebIdentityCredentialsUtil.factory();
+            credentialsProvider = factory.create(roleArn, roleSessionName, webIdentityToken);
```

Comment:
Is this a closeable credentials provider? Do we need to make this provider closeable as well?